sote/game/options.lua

```diff
@@ -7,7 +7,7 @@ local opt = {}
 function opt.init()
 	return {
 		["volume"] = 0,
-		["fullscreen"] = true,
+		["fullscreen"] ="normal",
```

Comment:
Standalone strings arent type safe for de facto enums, it needs a type declaration for the LSP

Reply:
I don't know exactly what you want here as I have only been working with lua for a week now and only understand it through your code and quick web searches. Is this just annotation for the language server on the function or variable? Some kind of reference or example would be helpful.

Reply:
Ah, sorry, I should have been more specific.

Wrt your edit, thats one way to do it, yeah. 
Though, I think in SotE the most common way is with aliases as enums as that avoids having to hope for the JIT to inline table lookup:
```lua
--@alias enumname value_a | value_b | value_c
```

You'd then need to annotate the function to mark the parameter of the type as `enumname`. See `function Rect:subrect` in `sote/engine/ui.lua` for an example of annotating a function with parameter types (in particular, enums).

SotE uses this LSP: https://github.com/LuaLS/lua-language-server
Its highly recommended to get an extension for it for your IDE